test/test-threadpool-cancel.c

```diff
@@ -184,12 +184,19 @@ TEST_IMPL(threadpool_cancel_getaddrinfo) {
   r = uv_getaddrinfo(loop, reqs + 3, getaddrinfo_cb, "fail", NULL, &hints);
   ASSERT(r == 0);
 
+#if defined(__APPLE__) && defined(_USE_ASYNC_ADDR_INFO)
+  # define TIMEOUT 0
+#else
+  # define TIMEOUT 10
+#endif
+
   ASSERT(0 == uv_timer_init(loop, &ci.timer_handle));
-  ASSERT(0 == uv_timer_start(&ci.timer_handle, timer_cb, 10, 0));
+  ASSERT(0 == uv_timer_start(&ci.timer_handle, timer_cb, TIMEOUT, 0));
```

Comment:
In case async addr is enable, the "task" is not considered as low prio anymore. Therefore the timer might be scheduled after uv_getaddrinfo completes. In order to properly cancel the task a timer with timeout 0 is created.

Reply:
I think I follow your line of reasoning. This test saturates the thread pool but the request is not going through the thread pool anymore.

I'm okay with changing the timeout to 0 unconditionally, no compile-time guards, as long as you add a comment explaining why it's 0.

Reply:
Yeah, you got my idea in here. I'll leave with the timeout set to 0.